Sources/ComposableArchitecture/Documentation.docc/Articles/SharingState.md

```diff
@@ -0,0 +1,539 @@
+  }
+  // ...
+}
+```
+
+When the parent features creates the child feature's state, it can pass a _reference_ to the shared
+count rather than the actual count value by using the `$count` projected value:
+
+```swift
+case .presentButtonTapped:
+  state.child = ChildFeature.State(count: state.$count)
+  // ...
+```
+
+Now any mutation the `ChildFeature` makes to its `count` will be instantly made to the 
+`ParentFeature`'s count too.
+
+## Persisted shared state
+
+Explicit shared state discussed above is a nice, lightweight way to share a piece of data with
+many parts of your application. However, sometimes you want to share state with the entire 
+application without having to pass it around explicitly. One can do this by passing a
+``PersistenceKey`` to the `@Shared` property wrapper, and the library comes with three persistence
+strategies, as well as the ability to create custom persistence strategies.
+
+#### In-memory
+
+This is the simplest persistence strategy in that it doesn't actually persist at all. It keeps
+the data in memory and makes it available to every part of the application, but when the app is
+relaunched the data will be reset back to its default.
+
+It can be used by passing ``PersistenceKey/inMemory(_:)`` to the `@Shared` property wrapper. For
+example, suppose you want to share an integer count value with the entire application so that any
+feature can read from and write to the integer. This can be done like so:
+
+```swift
+@Reducer
+struct ChildFeature {
+  @ObservableState
+  struct State {
+    @Shared(.inMemory("count")) var count = 0
+    // Other properties
+  }
+  // ...
+}
+```
+
+> Note: When using a persistence strategy with `@Shared` you must provide a default value, which is
+> used for the first access of the shared state.
+
+Now any part of the application can read from and write to this state, and features will never
+get out of sync.
+
+#### User defaults
+
+If you would like to persist your shared value across application launches, then you can use the
+``PersistenceKey/appStorage(_:)-9zd2f`` strategy with `@Shared` in order to automatically persist
+any changes to the value to user defaults. It works similarly to in-memory sharing discussed above,
+but it requires a key to store the value in user defaults, as well as a default value that will be
+used when there is no value in the user defaults:
+
+```swift
+@Shared(.appStorage("count")) var count = 0
+```
+
+That small change will guarantee that all changes to `count` are persisted and will be 
+automatically loaded the next time the application launches.
+
+This form of persistence only works for simple data types because that is what works best with
+`UserDefaults`. This includes strings, booleans, integers, doubles, URLs, data, and more. If you
+need to store more complex data, such as custom data types serialized to JSON, then you will want
+to use the <doc:SharingState#File-storage> strategy or a <doc:SharingState#Custom-persistence>
```

Comment:
This is showing up like this:
![Screenshot 2024-02-27 at 15 46 14](https://github.com/pointfreeco/swift-composable-architecture/assets/80204376/212be12b-2bc6-4518-b12d-831b3b86c0c5)

Reply:
Is that screenshot from GitHub or from Xcode docs? It’s only meant to render properly in Xcode as it’s Docc syntax.

Reply:
Ahh. My bad. That was GitHub

Reply:
Just noticed this says "then you will want to use the Sharing state strategy or a Sharing state strategy." Should it not say.

```
then you will want to use the File storage strategy or a custom persistence strategy
```

Screenshot from Xcode Docs 

<img width="629" alt="Screenshot 2024-02-28 at 10 44 28" src="https://github.com/pointfreeco/swift-composable-architecture/assets/80204376/d2aea261-b2cf-4ede-a885-eaa74565a2be">

Reply:
What version of Xcode are you using and how are you viewing the docs? When the docs are fully built it will show correctly. For example, here is what it looks like for me when I build docs (cmd+ctrl+shift+D):

<img width="746" alt="Screenshot 2024-02-28 at 4 20 28 PM" src="https://github.com/pointfreeco/swift-composable-architecture/assets/135203/719e5d81-b224-41dd-a3a8-f4fd02466e53">

But for whatever reason the preview of docs in Xcode does not expand those names. So I don't think this is actually an issue when docs are generated.

Reply:
I was using Xcode 15.2. And yes, it was previews.